pkg/scheduler/plugins/util/nodelock/nodelock_test.go

```diff
@@ -0,0 +1,61 @@
+package nodelock
+
+import (
+	"fmt"
+	"reflect"
+	"testing"
+	"time"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes/fake"
+)
+
+func TestLockNode(t *testing.T) {
+	var (
+		nodeName = "test-node"
+		lockName = "test-node-lock"
+	)
+
+	tests := []struct {
+		name      string
+		expectErr error
+		node      *v1.Node
+	}{
+		{
+			name:      "lock node success",
+			expectErr: nil,
+			node: &v1.Node{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: nodeName,
+					Annotations: map[string]string{
+						lockName: time.Now().Add(-time.Minute * 20).Format(time.RFC3339),
+					},
+				},
+			},
+		},
+		{
+			name:      "lock node failed",
+			expectErr: fmt.Errorf("node %s has been locked within 5 minutes", nodeName),
+			node: &v1.Node{
```

Comment:
How about use `buildNode` ?

Reply:
`import cycle not allowed in test`
If we use the `util.BuildNode`, there will be an import cycle not allowed problem, so I use `buildNode`